sony.go

```diff
@@ -0,0 +1,221 @@
+
+	HTTPClient *http.Client
+}
+
+var (
+	// ErrHTTP is a sentinel error for all HTTP status-based errors, It is
+	// intended to be used with `errors.Is(err, ErrHTTP)`.
+	ErrHTTP = errors.New("http")
+
+	// ErrSony is a sentinel error for errors returned by the REST IP
+	// control protocol in the body of a response.
+	ErrSony = errors.New("sony")
+
+	// ErrInvalidSonyResponse is a sentinel error for when a SonyError
+	// could not be parsed from the JSON response when an error is returned.
+	// See NewSonyError() for details.
+	ErrInvalidSonyResponse = errors.New("invalid Sony error response")
+)
+
+// HTTPStatusError captures the status code of a HTTP response that is to be
+// treated as an error.
+type HTTPStatusError int
+
+// Error formats a HTTP status code as its text description as per
+// http.StatusText().
+func (err HTTPStatusError) Error() string {
+	return http.StatusText(int(err))
+}
+
+// Unwrap returns ErrHTTP for all errors so HTTPStatusErrors can be checked
+// with `errors.Is(err, ErrHTTP)`, instead of the two-line
+// `var errHTTP HTTPStatusErr; errorsAs(err, &errHTTP)`. The latter is still
+// possible if the error type is required.
+func (err HTTPStatusError) Unwrap() error {
+	return ErrHTTP
+}
+
+// SonyError captures an error returned by the Sony REST IP control protocol
+// as an error returned in the payload of an HTTP response. These errors are
+// returned as an error code and a string describing it.
+type SonyError struct {
+	Code    int
+	Message string
+}
+
+// NewSonyError returns a SonyError from the response. The []any parameter
+// is expected to have two elements, an int (code) and a string (message).
+// If the size or types are not as just described, ErrInvalidSonyResponse
+// is returned.
+func NewSonyError(resp []any) error {
+	if len(resp) != 2 {
+		return ErrInvalidSonyResponse
+	}
+	code, ok := resp[0].(int)
+	if !ok {
+		return ErrInvalidSonyResponse
+	}
+	msg, ok := resp[1].(string)
+	if !ok {
+		return ErrInvalidSonyResponse
+	}
+	return SonyError{Code: code, Message: msg}
+}
+
+// Error returns the message portion of the error.
+func (err SonyError) Error() string {
+	return err.Message
+}
+
+// Unwrap returns ErrSony for all errors so SonyErrors can easily be checked
+// with `errors.Is(err, ErrSony)` for when you don't care about the specific
+// code or message of the error (when you would use `errors.As()`).
+func (err SonyError) Unwrap() error {
+	return ErrSony
+}
+
+// NewRESTClient creates and returns a BraviaClient reachable at the given
+// hostname, using the Pre-Shared Key given as psk as the password. If psk is
+// the empty string, it is not used.
+func NewRESTClient(hostname, psk string) *RESTClient {
+	return &RESTClient{
+		BaseURL:    "http://" + hostname + "/sony",
+		PSK:        psk,
+		HTTPClient: &http.Client{},
+	}
+}
+
+// none is a type to be used with `post[T]()` for when a response is not returned.
+// e.g. `_, err := post[none](...)`.
+type none struct{}
```

Comment:
have you considered `empty`?

Reply:
Yep. Was on the fence between `none` and `empty`. Will change to `empty`. Thanks.

Reply:
Done in e865205e2746d05ef1a1d258ce5f9fbbb9bb2f20